articles/architecture-scenarios/b2c/user-provisioning.md

```diff
@@ -0,0 +1,83 @@
+
+### Use Auth0 as the identity data store
+
+Auth0 provides identity data storage to help you manage the burden of storing user credentials safely and securely. If you have a legacy identity store that you would like Auth0 to host, you can use Auth0’s user migration capabilities. 
+
+## User migration
+
+If you decide to use Auth0 instead of your own identity store, you can [migrate users](/users/concepts/overview-user-migration) with bulk migration or progressively with automatic migration. You can also choose a two-phased approach to user migration, using automatic migration first, then using bulk migration for the users that remain. For other scenarios, see [User Migration Scenarios](/users/references/user-migration-scenarios).
+
+### Bulk migration
+
+Auth0 recommends using the [Management API](/users/concepts/overview-user-migration#bulk-user-imports-with-the-management-api) for bulk migration over the [User Import/Export extension](https://auth0.com/docs/users/concepts/overview-user-migration#migrate-users-with-the-user-import-export-extension) in all but the most simple cases. The Management API provides more flexibility and control. 
+
+After the bulk migration is complete, your users will need to reset their passwords.
+
+### Automatic migration
+
+[Automatic migration](/users/guides/configure-automatic-migration) allows users to be migrated incrementally, so they will not need to reset their passowrds. 
+
+## Self Sign Up
+
+Self sign up leverages Auth0 database connections to store the user id, password and username collected during the sign up process. Using the Auth0 Dashboard, you can configure the database connection policies such as minimum username length and password options such as password strength and complexity.  
+
+Auth0 provides a widget, [Lock](/libraries), which provides user interface functionality for sign up out-of-the-box. Lock supports a range of customization for user self sign up as well as login, however if you need to make extensive changes to CSS, Javascript or HTML to achieve your branding goals then you should use a fully customized [Universal Login](/universal-login) instead. For a comparison of the capabilities, see [Lock vs. a Custom UI](/libraries/when-to-use-lock).
+
+## Best practices
+
+* For information on the best ways to store your user data, see [User Data Storage Best Practices](/best-practices/user-data-storage-best-practices). 
+
+* Using a web based workflow with [Auth0 Universal Login](/universal-login) for sign up is considered both industry and Auth0 best practice because it provides optimal functionality and security. 
+
+* Calls to the Management API are subject to [Auth0 Rate Limiting policy](/docs/policies/rate-limits). Auth0 recommends using the [Auth0 SDK](/libraries) for your development environment rather than calling our APIs directly. 
```

Comment:
Calls to the Management API are subject to [rate limits](/docs/policies/rate-limits). Auth0 recommends using the appropriate [SDK](/libraries) for your development environment, rather than calling the APIs directly.